### ISSUE TITLE
Error possibly related to bibliography parsing

### ISSUE THREAD

**Issue by jonhanke** [plastex/plastex#126]
I'm fairly new to this and trying to convert a few papers with plasTeX on the command line.  There seems to be an issue when trying to convert `*.tex` file of the paper [here](https://arxiv.org/abs/1010.1926) (from the source `*.tar.gz` tarball available [here](https://arxiv.org/e-print/1010.1926)).  The command 

```
plastex Berkovich-Jagy_proof__re-submitted_to_Arxiv_2011-04-13_.tex
```

in that directory gives a few errors, with the last one ending in 
```
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```
seeming to be related to the plasTeX bibliography parsing.  The directory unpacks to contain one `*.tex` and one `*.bbl` file.  

I've also just `pip install`-ed the latest version of this repository, thanks to the helpful advice resolving issue #125.

Any help understanding/resolving this is appreciated.

Thanks!
-Jon

<hr>
<br>
<b> Error Message: </b>

```
...
ERROR: Error while expanding "bibliography" in <tokens> on line 1
   (unsupported operand type(s) for +: 'NoneType' and 'str')
ERROR: An error occurred while building the document object in <tokens> on
   line 1 (unsupported operand type(s) for +: 'NoneType' and 'str')
ERROR: Error while reading argument "len" of \ in <string> on line 1487
   (unsupported operand type(s) for +: 'NoneType' and 'str')
ERROR: Error while parsing argument "len" of "\"
ERROR: Error while expanding "\" in <string> on line 1487 (unsupported
   operand type(s) for +: 'NoneType' and 'str')
ERROR: An error occurred while building the document object in <string> on
   line 1487 (unsupported operand type(s) for +: 'NoneType' and 'str')
Traceback (most recent call last):
  File "/Users/jonhanke_devel/anaconda3/bin/plastex", line 129, in <module>
    main(sys.argv)
  File "/Users/jonhanke_devel/anaconda3/bin/plastex", line 73, in main
    tex.parse()
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py", line 441, in parse
    item.digest(tokens)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/__init__.py", line 905, in digest
    item.digest(tokens)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/Base/LaTeX/Sectioning.py", line 295, in digest
    item.digest(tokens)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/Base/LaTeX/Definitions.py", line 101, in digest
    Environment.digest(self, tokens)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/__init__.py", line 905, in digest
    item.digest(tokens)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/__init__.py", line 905, in digest
    item.digest(tokens)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/__init__.py", line 889, in digest
    for item in tokens:
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py", line 46, in __next__
    return self._next()
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py", line 346, in __iter__
    tokens = obj.invoke(self)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/__init__.py", line 363, in invoke
    self.parse(tex)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/__init__.py", line 468, in parse
    **arg.options)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py", line 760, in readArgumentAndSource
    toks, source = self.readGrouping(spec, expanded, parentNode=parentNode)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py", line 912, in readGrouping
    toks = self.expandTokens(toks, parentNode=parentNode)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py", line 409, in expandTokens
    out = tex.parse(frag)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py", line 438, in parse
    for item in tokens:
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py", line 46, in __next__
    return self._next()
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py", line 346, in __iter__
    tokens = obj.invoke(self)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/Base/LaTeX/Bibliography.py", line 23, in invoke
    self.loadBibliographyFile(tex)
  File "/Users/jonhanke_devel/anaconda3/lib/python3.7/site-packages/plasTeX/Base/LaTeX/Bibliography.py", line 29, in loadBibliographyFile
    filename = tex.kpsewhich(tex.jobname+'.bbl')
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```

**Comment by PatrickMassot** [comment]
I'm sorry I had difficulties finding time to download you source file and play with it. I have a couple of remarks that could help you to get moving.
* The very first thing you want to do when using plasTeX on a document which was not written with that compiler in mind, is to make sure your preamble isn't bloated with exotic packages you used ten years earlier and copy-paste to each new paper. In your case, this means `epstopdf` (and the following line which is presumably a configuration command for that package).
* Then you need to think about whether you really need the exotic packages you do use. Here it means `cancel`. Clearly this has nothing to do with plasTeX: if mathjax doesn't support it then you are screwed. Here of course I assume you want to use mathjax, not small pictures from the 90's. In this case the "solution" is pretty simple: replace `\usepackage{cancel}` by `\newcommand\cancel[1]{#1}`
* At this stage the preamble looks rather clean, and you can run `plastex --renderer=HTML5 --dollars Berkovich_..._.tex` and plasTeX will actually do something.
* I see plasTeX seems to struggle with `$... \text{$S$-genus}$`. I dont' know what happens here (I'm not the author of plasTeX), but at least you can somehow fix this by using `$... S-\text{genus}$`.
* You also have mysterious `\[`, `\]`, `\(`, and `\)` inside math mode. I have no idea how that can be valid TeX, and I don't enough have courage to ask Google. Replacing them by `\left[` etc. will certainly help plasTeX.

After all those steps I hope the document starts being recognizable.

**Comment by PatrickMassot** [comment]
The `$... \text{$S$-genus}$` issue should be fixed by d17a553.
